lab4/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,61 @@
+package agh.ics.oop;
+
+import java.util.Objects;
+
+class Animal {
+    private MapDirection orientation = MapDirection.NORTH;
+    private Vector2d position;
+
+    private final IWorldMap worldMap;
+
+    public Animal() {
+        this(new RectangularMap(10, 10), new Vector2d(2, 2));
```

Comment:
w sumie nie mówiliśmy o domyślnej mapie, ale w takim wariancie też jest spoko przy czym z punktu widzenia ładności kodu: lepiej wywołać tu `this(new RectangularMap(10, 10))`, takie zagnieżdżanie konstruktorów zmniejsza redundancję danych w kodzie